src/java.base/share/classes/jdk/internal/foreign/AbstractMemorySegmentImpl.java

```diff
@@ -653,27 +632,25 @@ public static void copy(MemorySegment srcSegment, ValueLayout srcLayout, long sr
                             Object dstArray, int dstIndex,
                             int elementCount) {
 
-        long baseAndScale = getBaseAndScale(dstArray.getClass());
+        var dstInfo = Utils.BaseAndScale.of(dstArray);
```

Comment:
This is a case where I personally wouldn't use `var` because RHS is not clear, I have no idea what type dstInfo is.